### PR TITLE
Use single thread for pytest to maintain pure env

### PR DESCRIPTION
**Pull request by sudo-panda**

### CODE REVIEW

**Comment by sudo-panda** [comment]
No need to merge if not required. I am using this because running them on my system is too slow.

**Comment by vgvassilev** [comment]
The PR seems a rebased version of #18

**Comment by sudo-panda** [comment]
We can rebase and merge #18 when you have the time.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -212,6 +212,7 @@ jobs:
         python -m pip install --upgrade pip
         python -m pip install pytest
         python -m pip install pytest-xdist
+        find . -type f -iname "*.py" -exec sed -i '/@mark.xfail/d' {} \;
```
Comment (vgvassilev) [inline .github/workflows/ci.yml:215]: We can use the `--runxfail` option.

**Comment by vgvassilev** [comment]
@sudo-panda do we still need this PR?

**Pull request closed without merging**